packages/account/src/BaseAccount.ts

```diff
@@ -40,99 +23,122 @@ export abstract class SmartAccount implements ISmartAccount {
+      }
+    }
+   // If all necessary properties are present in userOp. we will skip estimation and return userOp
+    if (skipEstimations) {
+      return userOp
+    }
+
     if (!this.bundler) {
       if (!this.provider) throw new Error('Provider is not present for making rpc calls')
-      // if no bundler url is provided run offchain logic to assign following values of user
+      // if no bundler url is provided run offchain logic to assign following values of UserOp
       // maxFeePerGas, maxPriorityFeePerGas, verificationGasLimit, callGasLimit, preVerificationGas
       const feeData = await this.provider.getFeeData()
-      if (EIP1559_UNSUPPORTED_NETWORKS.includes(this.chainId)) {
-        // assign gasPrice to both maxFeePerGas and maxPriorityFeePerGas in case chain does not support Type2 transaction
-        userOp.maxFeePerGas = userOp.maxPriorityFeePerGas =
-          feeData.gasPrice ?? (await this.provider.getGasPrice())
-      } else {
-        if (!feeData.maxFeePerGas || !feeData.maxPriorityFeePerGas)
-          throw new Error('Unable to get maxFeePerGas and maxPriorityFeePerGas from provider')
-        userOp.maxFeePerGas = feeData.maxFeePerGas
-        userOp.maxPriorityFeePerGas = feeData.maxPriorityFeePerGas
-      }
-      userOp.verificationGasLimit = await this.getVerificationGasLimit(userOp.initCode)
-      userOp.callGasLimit = await this.provider.estimateGas({
+      userOp.maxFeePerGas = userOp.maxFeePerGas ?? feeData.maxFeePerGas ?? (await this.provider.getGasPrice())
+      userOp.maxPriorityFeePerGas = userOp.maxPriorityFeePerGas ?? feeData.maxPriorityFeePerGas ?? (await this.provider.getGasPrice())
+      if (userOp.initCode)
+        userOp.verificationGasLimit = userOp.verificationGasLimit ?? await this.getVerificationGasLimit(userOp.initCode)
+      userOp.callGasLimit = userOp.callGasLimit ?? await this.provider.estimateGas({
         from: this.smartAccountConfig.epAddress,
         to: userOp.sender,
         data: userOp.callData
       })
-      userOp.preVerificationGas = this.getPreVerificationGas(userOp)
+      userOp.preVerificationGas = userOp.preVerificationGas ?? this.getPreVerificationGas(userOp)
     } else {
       // Making call to bundler to get gas estimations for userOp
-      const gasEstimationResponse = await this.bundler.getUserOpGasFields(userOp, this.chainId)
       const {
         callGasLimit,
         verificationGasLimit,
         preVerificationGas,
-        gasPrice,
         maxFeePerGas,
         maxPriorityFeePerGas
-      } = gasEstimationResponse.result
-      if (gasPrice) userOp.maxFeePerGas = userOp.maxPriorityFeePerGas = gasPrice
+      } = await this.bundler.getUserOpGasFields(userOp, this.chainId)
+      if ((!userOp.maxFeePerGas && !userOp.maxPriorityFeePerGas) && (!maxFeePerGas || !maxPriorityFeePerGas)) {
+        const feeData = await this.provider.getFeeData()
+        userOp.maxFeePerGas = feeData.maxFeePerGas ?? (await this.provider.getGasPrice())
```

Comment:
resolved

Reply:
since gasPrice return from FeeData could be null so for nullish check calculated it from provider object. since its fallback so will run only in case gasPrice returned from FeeData is null.

`userOp.maxFeePerGas = userOp.maxFeePerGas ?? feeData.maxFeePerGas ?? feeData.gasPrice ?? (await this.provider.getGasPrice())`